api/middleware/middleware.go

```diff
@@ -49,7 +49,11 @@ func TokenMiddleware(nextHandler http.Handler) http.Handler {
 			visas, err := auth.GetVisas(auth.Details, token)
 			if err != nil {
 				log.Debug("failed to validate token at AAI")
-				http.Error(w, "bad token", 401)
+				c.String(http.StatusUnauthorized, "bad token")
+				err := c.AbortWithError(code, err)
+				if err != nil {
+					log.Errorf("Error: %v", err)
+				}
```

Comment:
```suggestion
				c.String(http.StatusUnauthorized, err.Error())
				c.AbortWithStatus(code)
```

Propagate the real error message as the token can be ok but the user might not have any visas.

Reply:
This shouldn't trigger when there are no visas though? or does it? should just get an empty visa list?

Reply:
I like `c.AbortWithStatus(code)` though! :) much cleaner!

Reply:
Yes, this only happens if the request to AAI fails (token is bad, or AAI response is malformed). And yes, we should proceed with an empty visa list, because authentication was successful, but if there are no visas, authorization will fail in a later phase.